tests/test_healpix_mapper.py

```diff
@@ -9,6 +12,24 @@
 
 class TestOctahedralGrid:
     def setup_method(self, method):
+        nexus_url = "https://get.ecmwf.int/test-data/polytope/test-data/healpix.grib"
+
+        local_directory = "./tests/data"
+
+        if not os.path.exists(local_directory):
+            os.makedirs(local_directory)
+
+        # Construct the full path for the local file
+        local_file_path = os.path.join(local_directory, "healpix.grib")
+
+        if not os.path.exists(local_file_path):
+            session = requests.Session()
+            response = session.get(nexus_url)
+            if response.status_code == 200:
+                # Save the downloaded data to the local file
+                with open(local_file_path, "wb") as f:
+                    f.write(response.content)
+
```

Comment:
This code is copy-pasted 6 times and so should be refactored into a helper function. Looks like the helper only needs 1 input argument, the `nexus_url` - the `local_directory` is hard-coded to the same location in every case and the `local_file_path` is derived from the `nexus_url` in every case.